ga4gh/datamodel/reads.py

```diff
@@ -335,66 +343,73 @@ class HtslibReadGroupSet(AlignmentDataMixin, AbstractReadGroupSet):
     """
     Class representing a logical collection ReadGroups.
     """
-    def __init__(
-            self, parentContainer, localId, samFilePath, dataRepository):
+    defaultReadGroupName = "default"
+
+    def __init__(self, parentContainer, localId):
         super(HtslibReadGroupSet, self).__init__(parentContainer, localId)
-        self._samFilePath = samFilePath
-        samFile = self.getFileHandle(self._samFilePath)
+        self._programs = []
+        self._dataUrl = None
+        self._indexFile = None
+        # Used when we populate from a file. Not defined when we populate
+        # from the DB.
+        self._bamHeaderReferenceSetName = None
+
+    def getBamHeaderReferenceSetName(self):
+        """
+        Returns the ReferenceSet name using in the BAM header.
+        """
+        return self._bamHeaderReferenceSetName
+
+    def populateFromRow(self, row):
+        """
+        Populates the instance variables of this ReadGroupSet from the
+        specified database row.
+        """
+        self._dataUrl = row[b'dataUrl']
+        self._indexFile = row[b'indexFile']
+        self._programs = []
```

Comment:
I don't think so because we want a referenceSet instance not the ID. I guess we could pass the repo instance through to ``populateFromRow`` and get the referenceSet from there, but this has it's own issues. I'm not overly happy with this ``populateFromRow`` design in any case because we're splitting the knowledge about the SQL schema between several modules. Hopefully we can think of something neater when we're redesigning this container layer.